crates/optimism/rpc/src/sequencer.rs

```
@@ -70,6 +70,37 @@ impl SequencerClient {
         }
     }
 
+    /// Creates a new `SequencerClient` for the given URL with additional custom headers.
+    pub async fn new_with_headers(
+        sequencer_endpoint: impl Into<String>,
+        headers: Vec<String>,
+    ) -> Result<Self, Error> {
+        let sequencer_endpoint = sequencer_endpoint.into();
+        let endpoint = BuiltInConnectionString::from_str(&sequencer_endpoint)?;
+        if let BuiltInConnectionString::Http(url) = endpoint {
+            let mut builder = reqwest::Client::builder().use_rustls_tls();
+
+            if !headers.is_empty() {
+                let mut header_map = reqwest::header::HeaderMap::new();
+                for header in headers {
+                    let (key, value) = header.split_once('=').unwrap();
+                    header_map.insert(
+                        key.trim().parse::<reqwest::header::HeaderName>().unwrap(),
+                        value.trim().parse::<reqwest::header::HeaderValue>().unwrap(),
+                    );
```

these can't use unwrap and should return an error instead